File: androidApp/src/main/kotlin/subha/app/cyra/ui/components/CyraSegmentedToggle.kt
package subha.app.cyra.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.interaction.MutableInteractionSource
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.remember
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import subha.app.cyra.ui.theme.CyraTheme

/**
 * A pill-shaped 2-4 option switch - the height/weight unit toggle (cm/ft-in, kg/lb) and
 * the "How is your cycle?" 3-way choice (Regular/Irregular/Not sure) are both this same
 * component, just with a different [options] list. Not a Boolean-only toggle - stores
 * the selected index so a third option (cycle regularity) needs no separate component.
 */
@Composable
fun CyraSegmentedToggle(
    options: List<String>,
    selectedIndex: Int,
    onOptionSelected: (Int) -> Unit,
    modifier: Modifier = Modifier,
) {
    Row(
        modifier = modifier
            .background(color = MaterialTheme.colorScheme.surfaceVariant, shape = RoundedCornerShape(12.dp))
            .padding(3.dp),
    ) {
        options.forEachIndexed { index, label ->
            val selected = index == selectedIndex
            Text(
                text = label,
                style = MaterialTheme.typography.labelMedium,
                color = if (selected) MaterialTheme.colorScheme.onPrimary else MaterialTheme.colorScheme.onSurfaceVariant,
                textAlign = TextAlign.Center,
                modifier = Modifier
                    .weight(1f)
                    .clickable(
                        interactionSource = remember { MutableInteractionSource() },
                        indication = null,
                        onClick = { onOptionSelected(index) },
                    )
                    .then(
                        if (selected) {
                            Modifier.background(color = MaterialTheme.colorScheme.primary, shape = RoundedCornerShape(10.dp))
                        } else {
                            Modifier
                        },
                    )
                    .padding(vertical = 10.dp, horizontal = 12.dp),
            )
        }
    }
}

@CyraPreviews
@Composable
private fun CyraSegmentedTogglePreview() {
    CyraTheme {
        CyraSegmentedToggle(options = listOf("Regular", "Irregular", "Not sure"), selectedIndex = 0, onOptionSelected = {})
    }
}

File: iosApp/iosApp/Components/CyraSegmentedToggle.swift
import SwiftUI

/// A 2-4 option switch - mirrors Android's `CyraSegmentedToggle.kt`. The height/weight
/// unit toggle (cm/ft-in, kg/lb) and the "How is your cycle?" 3-way choice
/// (Regular/Irregular/Not sure) are both this same view, just with a different
/// `options` list. Not a Boolean-only toggle - stores the selected index so a third
/// option (cycle regularity) needs no separate view.
///
/// Each option is its own pill: unselected options show an outline only (no fill), the
/// selected option shows a solid fill (no border) - a first draft had unselected
/// options as plain, borderless text, which didn't read as tappable.
struct CyraSegmentedToggle: View {
    let options: [String]
    let selectedIndex: Int
    let onOptionSelected: (Int) -> Void

    var body: some View {
        HStack(spacing: 8) {
            ForEach(Array(options.enumerated()), id: \.offset) { index, label in
                let selected = index == selectedIndex
                Text(label)
                    .font(CyraFont.labelMedium())
                    .foregroundColor(selected ? .cyraOnPrimary : .cyraOnSurfaceVariant)
                    .frame(maxWidth: .infinity)
                    .padding(.vertical, 10)
                    .background(
                        RoundedRectangle(cornerRadius: 10)
                            .fill(selected ? Color.cyraPrimary : Color.clear),
                    )
                    .overlay(
                        RoundedRectangle(cornerRadius: 10)
                            .stroke(selected ? Color.clear : Color.cyraOutline, lineWidth: 1),
                    )
                    .onTapGesture { onOptionSelected(index) }
            }
        }
    }
}

#Preview {
    CyraSegmentedToggle(options: ["Regular", "Irregular", "Not sure"], selectedIndex: 0, onOptionSelected: { _ in })
        .padding(24)
        .cyraThemed()
}
